### DISCUSSION TITLE
Rewrote it in Rust!

### DISCUSSION THREAD

**Discussion by dhruvkb** [pls-rs/pls#78] [Announcements]
Over the last few weeks, I've rewritten `pls` in Rust. This was motivated by a number of factors, not the least of which is that Rust is an interesting, new language that I wanted to learn.

Hence [v6.0.0](https://pypi.org/project/pls/6.0.0/) will be the last Python-based release of `pls`. It contains all the code that has been pushed to the `main` branch so far.

The Python-based iteration of `pls` has served us well and I'm very fond of the codebase. So I will continue to host the Python codebase at the `pls-py` branch for the foreseeable future. `pls` will also continue to be available at PyPI and installable via `pip`/`pipx`. All installed copies of `pls` will continue to work but they will not receive any further upgrades.

The Rust-based iteration of `pls` has been in the works for several weeks and is fairly stable now. As I write the docs for it and apply finalising touches like setting up a pipeline for releases and distribution, I want to bring it out of stealth mode to build in the open.  This repo will now host the Rust codebase of `pls`.

I hope all users of `pls` can migrate to the Rust-based `pls` soon. See you on the other side.

**Comment by djhunter67** [comment]
Be advised, as long the the rust code-base is compiled for the architecture of your computer the execution is ``./pls`` instead of ``python pls``.
